jet_tracking/jet_tracking_cal/jt_cal.py

```diff
@@ -0,0 +1,441 @@
+    left_line = Span(location=low, dimension='height', \
+        line_color='black')
+    right_line = Span(location=high, dimension='height', \
+        line_color='black')
+    peak_line = Span(location=med, dimension='height', \
+        line_color='red')
+    fig.renderers.extend([left_line, right_line, peak_line])
+
+    return fig
+
+def azav_fig(ave_azav, peak, intensity, delta_bin):
+    """Generate the azav fig for html file"""
+    x_vals = np.arange(len(ave_azav))
+    fig = figure(
+        title='Average Azimuthal Binned Array: Center - {0}, \
+            min/max - {1}/{2}, intensity - {3}'.format(peak, \
+            peak-delta_bin, peak+delta_bin, round(intensity, 2)),
+        x_axis_label='Bins',
+        y_axis_label='Intensity',
+    )
+
+    peak_line = Span(location=peak, dimension='height', \
+        line_color='green', line_width=2)
+    lower_line = Span(location=peak-delta_bin, dimension='height', \
+        line_color='black')
+    upper_line = Span(location=peak+delta_bin, dimension='height', \
+        line_color='black')
+    ave_azav_curve = fig.scatter(x_vals, ave_azav)
+    fig.renderers.extend([peak_line, lower_line, upper_line])
+
+    azav_legend = Legend(items=[
+        LegendItem(label='Azimuthal Average', renderers=[ave_azav_curve])
+    ])
+    fig.add_layout(azav_legend)
+
+    return fig
+
+def intensity_hist(intensity_hist, edges):
+    fig = figure(
+            title='Intensity Histogram'
+    )
+    fig.quad(top=intensity_hist, bottom=0, left=edges[:-1], right=edges[1:])
+
+    return fig
+
+def intensity_vs_peak_fig(intensity, peak_vals, x, y, slope, intercept, sigma):
+    """Simple plot of intensity vs peak value"""
+    fig = figure(
+        title='Peak value vs Intensity. Slope = {0}, Intercept = {1}'.\
+            format(round(slope, 2), round(intercept, 2)),
+        x_axis_label='Intensity Monitor Value',
+        y_axis_label='Peak Values'
+    )
+    fig.x_range.range_padding = fig.y_range.range_padding = 0
+    h, y_edge, x_edge = np.histogram2d(peak_vals, intensity, bins=100)
+    fig.image(image=[h], x=x_edge[0], y=y_edge[0], dh=y_edge[-1]-y_edge[0], \
+        dw=x_edge[-1]-x_edge[0], palette="Spectral11")
+    color_mapper = LinearColorMapper(palette="Spectral11", low=h.min(), high=h.max())
+    color_bar = ColorBar(color_mapper=color_mapper, location=(0,0))
+    fig.xaxis.bounds = [i0_low, i0_high]
+    fig.add_layout(color_bar, 'right')
+    fig.line(x, y, color='red')
+    fig.line(x, y - 1 * sigma, color='orange')
+    fig.line(x, y + 1 * sigma, color='orange')
+    return fig
+
+if __name__ == '__main__':
+    parser = ArgumentParser()
+    parser.add_argument('--exp', type=str, \
+        default=(os.environ.get('EXPERIMENT', None)))
+    parser.add_argument('--run', type=str, \
+        default=(os.environ.get('RUN_NUM', None)))
+    parser.add_argument('--cfg', type=str, \
+        default=(''.join([JT_LOC, 'mpi_scripts/mpi_configs/default_config.yml'])))
```

Comment:
Good for now. Todo for later: add some autodetection, sort of like happi does for its cfg